Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -70,6 +70,8 @@ WarpX::LoadBalance ()
     }
     if (doRedistribute)
     {
+        // Re-initialize diagnostic functors that stores pointers to the user-requested fields.
+        multi_diags->InitializeFieldFunctors();
```

Comment:
are the old functors destroyed at this point? Just checking, because they do hold quite an amount of dynamically allocated data members with `new` and I cannot quickly spot the `delete` counter parts.

Reply:
Good point.

Reply:
maybe `std::unique_ptr` is an option there.

Reply:
Thank you @ax3l for the great point and for sharing your knowledge on this offline. As per our discussion, I have included some changes in the PR. @mrowan137 that is right! I have now included std::unique_ptr  in this PR. 
Thanks all!